R/run.R

```diff
@@ -20,16 +20,68 @@ run.default <- function(x, ...) {
 }
 
 
+#' @param publish Flag for whether to publish results to a Conbench server. See
+#' "Environment Variables" section for how to specify server details. Requires
+#' the benchconnect CLI is installed; see [install_benchconnect()].
+#' @param run_name Name for the run. If not specified, will use `{run_reason}: {commit hash}`
+#' @param run_reason Required. Low-cardinality reason for the run, e.g. "commit" or "test"
+#'
+#' @section Environment Variables:
+#'
+#' - `CONBENCH_URL`: Required. The URL of the Conbench server with no trailing
+#' slash. For arrow, should be `https://conbench.ursa.dev`.
+#' - `CONBENCH_EMAIL`: The email to use for Conbench login. Only required if the
+#' server is private.
+#' - `CONBENCH_PASSWORD`: The password to use for Conbench login. Only required
+#' if the server is private.
+#' - `CONBENCH_PROJECT_REPOSITORY`: The repository name (in the format
+#' `org/repo`) or the URL (in the format `https://github.com/org/repo`).
+#' Defaults to `"https://github.com/apache/arrow"` if unset.
+#' - `CONBENCH_PROJECT_PR_NUMBER`: Recommended. The number of the GitHub pull
+#' request that is running this benchmark, or `NULL` if it's a run on the
+#' default branch
+#' - `CONBENCH_PROJECT_COMMIT`: The 40-character commit SHA of the repo being
+#' benchmarked. If missing, will attempt to obtain it from
+#' `arrow::arrow_info()$build_info$git_id`, though this may not be populated
+#' depending on how Arrow was built.
+#' - `CONBENCH_MACHINE_INFO_NAME`: Will override detected machine host name sent
+#' in `machine_info.name` when posting runs and results. Needed for cases where
+#' the actual host name can vary, like CI and cloud runners.
+#'
 #' @rdname run
 #' @export
-run.BenchmarkDataFrame <- function(x, ...) {
+run.BenchmarkDataFrame <- function(x, ..., publish = FALSE, run_name = NULL, run_reason = NULL) {
   # if already run (so no elements of `parameters` are NULL), is no-op
   x <- get_default_parameters(x, ...)
 
+  if (publish) {
+    github <- github_info()
+    if (is.null(run_name)) {
+      run_name <- paste(run_reason, github$commit, sep = ": ")
+    }
+    stopifnot(
+      "Results cannot be published without a `run_reason`!" = !is.null(run_reason)
+    )
```

Comment:
Should this be up about lines 59-61? I know functionally this will stop if there's no run_reason, but it's a little weird that we concatenate `run_name` from `run_reason` regardless if it is null or not, and then check it after that.